### PR TITLE
fix(workspace): make item actions menu use sidebar authoring

### PR DESCRIPTION
**Pull request by petrjasek**
refactor authoringWorkspace from controller into a service
so that it can be used also from activities and not only directives.

SD-2885

### CODE REVIEW

File: client/app/scripts/superdesk-authoring/authoring.js [hunk 1]
```diff
@@ -1798,11 +1799,10 @@
...
             templateUrl: 'scripts/superdesk-authoring/views/authoring-container.html',
-            require: '^sdAuthoringWorkspace',
-            scope: true,
-            link: function(scope, elem, attrs, workspaceCtrl) {
-                scope.$watch(workspaceCtrl.getItem, function(item) {
-                    scope.authoring.edit(item, true);
+            scope: {},
+            link: function(scope) {
+                scope.$watch(authoringWorkspace.getItem, function(item) {
+                    scope.authoring.edit(item, !authoringWorkspace.readonly);
```
Comment (plamut) [inline client/app/scripts/superdesk-authoring/authoring.js]: We might want do skip this call when `$watcher` is invoked for the first time (due to initialization) and nothing has actually changed yet?

File: client/app/scripts/superdesk-authoring/tests/authoring_spec.js [hunk 2]
```diff
@@ -1030,30 +1027,60 @@ describe('authoring actions', function() {
...
-        }));
-    });
+        authoringWorkspace.close();
+        expect(authoringWorkspace.item).toBe(null);
+        expect(authoringWorkspace.getItem()).toBe(null);
+        expect(superdeskFlags.flags.authoring).toBeFalsy();
+
+        authoringWorkspace.edit(item, true);
+        expect(authoringWorkspace.readonly).toBe(true);
+    }));
```
Comment (plamut) [inline client/app/scripts/superdesk-authoring/tests/authoring_spec.js]: Seems like this test case is trying to test to much (`edit()`, `close()`, then `edit()` again), IMO it should probably be split to a couple of smaller independent test cases.

File: client/app/scripts/superdesk-authoring/tests/authoring_spec.js [hunk 3]
```diff
@@ -1030,30 +1027,60 @@ describe('authoring actions', function() {
...
+
+        authoringWorkspace.close();
+        scope.$digest();
+        expect(iscope.authoring.item).toBe(null);
+        expect(iscope.authoring.state.opened).toBe(false);
+
+        authoringWorkspace.edit(item, true);
+        scope.$digest();
+        expect(iscope.authoring.item.lockIt).toBe(false);
+    }));
```
Comment (plamut) [inline client/app/scripts/superdesk-authoring/tests/authoring_spec.js]: Same remark as above (splitting the test case).

**Comment by plamut** [comment]
We need more refactorings like that, good work! :+1:

**Comment by petrjasek** [comment]
now including #1023
let's see what travis thinks about it

**Comment by ioanpocol** [comment]
:+1:

**Pull request merged**